icu4c/source/common/uresbund.cpp

```diff
@@ -2327,6 +2329,18 @@ ures_openFillIn(UResourceBundle *r, const char* path,
     ures_openWithType(r, path, localeID, URES_OPEN_LOCALE_DEFAULT_ROOT, status);
 }
 
+/**
+ * Same as ures_openDirect(), but uses the fill-in parameter and does not allocate a new bundle.
+ */
+U_INTERNAL void U_EXPORT2
+ures_openDirectFillIn(UResourceBundle *r, const char* path, const char* localeID, UErrorCode* status) {
+    if(U_SUCCESS(*status) && r == NULL) {
```

Comment:
Is this the intended logic? This will not return early if the status is a failure but you pass in null. Should it return but leave the error untouched?

Reply:
Also, should this use nullptr instead of null?

Reply:
After this "if" is a call to a function with the UErrorCode parameter. That function should check for failure before doing something interesting. No need to optimize for error cases.

Reply:
The API called below (`ures_openWithType`) will early return if `status` is a failure.
I didn't change to use nullptr since the file already uses NULL.

Reply:
I feel that code should either check for errors at the outer-most layer only or check everywhere. Not checking at the outermost layer and relying on behavior of functions you call leads to interesting bugs down the road as code changes. But resolving the conversation anyway since it's ultimately not my call.